client/client-multi/client-c/CMakeLists.txt

```diff
@@ -161,34 +134,74 @@ set(KAA_INCLUDE_DIRS
         ${KAA_SRC_FOLDER}
         ${KAA_THIRDPARTY_INCLUDE_DIR})
 
-# Builds Kaa libraries.
+# Build unit tests.
+include(${CMAKE_CURRENT_LIST_DIR}/listfiles/UnitTest.cmake)
 
-set(KAA_BUILD_STATIC kaac_s)
-add_library(kaac_o OBJECT ${KAA_SOURCE_FILES})
-target_include_directories(kaac_o PUBLIC ${KAA_INCLUDE_DIRS})
+# Build Kaa libraries.
+add_library(kaac ${KAA_SOURCE_FILES})
+target_include_directories(kaac PUBLIC ${KAA_INCLUDE_DIRS})
+target_link_libraries(kaac PRIVATE ${KAA_THIRDPARTY_LIBRARIES})
 
-add_library(${KAA_BUILD_STATIC} STATIC ${KAA_SRC_FOLDER} $<TARGET_OBJECTS:kaac_o>)
-target_link_libraries(${KAA_BUILD_STATIC} PRIVATE ${KAA_THIRDPARTY_LIBRARIES})
-target_include_directories(${KAA_BUILD_STATIC} PUBLIC ${KAA_INCLUDE_DIRS})
+# Disables Kaa library modules.
+if(WITH_EXTENSION_PROFILE)
+    message("PROFILE ENABLED")
 
-if(NOT DEFINED KAA_BUILD_STATIC_ONLY)
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -fPIC")
-    set(KAA_BUILD_SHARED kaac)
-    add_library(${KAA_BUILD_SHARED} SHARED ${KAA_SRC_FOLDER} $<TARGET_OBJECTS:kaac_o>)
-    target_include_directories(${KAA_BUILD_SHARED} PUBLIC ${KAA_INCLUDE_DIRS})
-    target_link_libraries(${KAA_BUILD_SHARED} ${KAA_THIRDPARTY_LIBRARIES})
+    include(${CMAKE_CURRENT_LIST_DIR}/src/extensions/profile/CMakeLists.txt)
```

Comment:
Let's check that it is possible to link extension libs conditionally without conditional inclusion them into the build. In another words, it can be possible to add libraries and their unit tests no matter what, but they will be compiled in kaac on user's demand.